testcases/pkcs11/attribute.c

```diff
@@ -138,11 +184,164 @@ CK_RV do_TestAttributes(void)
     else
         testcase_pass("Successfully verified updated attributes.");
 
+    /* Try to update attributes of an object which has CKA_MODIFIABLE=FALSE */
+    testcase_new_assertion();
+
+    /* create a public key object with CKA_MODIFIABLE=FALSE*/
+    rc = funcs->C_CreateObject(session, pub_template_no_modify, 7,
+                               &obj_handle_no_mod);
```

Comment:
We could introduce a length variable, like
CK_ULONG pub_template_no_modify_len = sizeof(pub_template_no_modify) / sizeof(CK_ATTRIBUTE);
and use it here instead of explicitly coding 7.